include/robot_interfaces/n_joint_robot_types.hpp

```diff
@@ -27,12 +30,16 @@ namespace robot_interfaces
  *
  * @tparam N Number of joints
  */
+class Loggable;
```

Comment:
Is this forward declaration really needed (loggable.hpp is included above)?
Also, please do not add anything between a definition (of `NJointRobotTypes` in this case) and its corresponding doxygen comment block.